packages/loopring_v3/contracts/amm/LoopringAmmPool.sol

```diff
@@ -80,9 +76,9 @@ contract LoopringAmmPool is
     )
     {
         require(_controller != IAmmController(0), "ZERO_ADDRESS");
-        controller = _controller;
-        assetManager = _assetManager;
-        joinsDisabled = _joinsDisabled;
+        state.controller = _controller;
+        state.assetManager = _assetManager;
+        state.joinsDisabled = _joinsDisabled;
```

Comment:
Because we always use the pools after a proxy, this will set the data inside the implementation contract (which will never get used) and so these values will still be `0` when actually used.

But do we want to set this data for each pool separately? Doesn't seem that useful to me and increases cost quite a bit both for pool creation (more data to store when creating each pool) and when using the pool (reading this data from storage for each join/exit, currently even sometimes when not used).

My thought process was that most pools will use the same values so may as well just hard-code them in the implementation contract so this data automatically set for each pool for free. They are also constants so free to read. If we would like to update these for the pools we can just deploy a new implementation contract with updated values and update the proxy.

Reply:
Alright.